processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -840,14 +862,27 @@ public boolean hasEmptyAccessibleContructor() {
      * @return a list of type arguments or null, if superclass was not found
      */
     public List<Type> determineTypeArguments(Class<?> superclass) {
-        if ( qualifiedName.equals( superclass.getName() ) ) {
+        return determineTypeArguments( superclass.getName() );
+    }
+
+    /**
+     * Searches for the given superclass and collects all type arguments for the given className.
+     * This is used for support with Java 8 streams, as must use the FQN in order to be able to compile
```

Comment:
This actually works. I only need to use FQN in the `TypeFactory`. I thought that it will fail as we use the FQN in the `Conversions`. However, I ran the integration tests and it seems OK. I will revert the changes to this.